### ISSUE TITLE
comment length greater than max_line_length results in violation

### ISSUE THREAD

**Issue by HassanJaveed84** [sqlfluff/sqlfluff#299]
Using `master` branch
Having this .sqlfluff file
```
[sqlfluff:rules:L016]
max_line_length = 80
```
and this sql file with a comment length > 80
```sql
--The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog.
SELECT * FROM some_table
```
shows a violation `L:   1 | P:  92 | L016 | Line is too long.`
And if you try to fix it:
```
==== finding fixable violations ====
==== no fixable linting violations found ====
```

**Comment by alanmcruickshank** [comment]
Yeah, it's a violation, but it's also not currently fixable. I've not implemented a way of correcting that problem.

What I'll do in the mean time is improve the logging around it however.

Do you have ideas about how this kind of issue should be corrected?

**Comment by alanmcruickshank** [comment]
@HassanJaveed84 - the easy way to fix this one would be to split on any whitespace and make it fit onto as many lines as it needs to to fit.

**Comment by aaronsteers** [comment]
Hi there. Was browsing through the tickets list and found this, which was of interest to me because I was dealing today with a URL in one of our comments which was about 50 or 60 characters long. I was able to resolve this by breaking the comment into two lines, with the URL on it's own line, but there of course can be URLs which are by themselves too long to fit under the line length.

I wonder if a good solution would be to add a configuration option to ignore line length for comments, while still respecting the line limit for all non-comment code. 

Another middle-ground option if #201 goes through, would be to have a lower severity (warning) for comments vs the core code itself. Meaning, we care about comment length enough to add yellow squiggles in the IDE, but not enough to block the CI/CD pipeline.

**Comment by pwildenhain** [comment]
Another option here would be to add `--noqa: L016` to ignore the long line

**Comment by barrywhart** [comment]
+1 for providing a way to ignore overly long comment lines. It's not unusual for comments to involve formatting such as:
* Paragraph breaks
* `-` or `*` or similar characters for bullet or number lists

**Comment by alanmcruickshank** [comment]
@barrywhart - with the merge of #751 - are you happy for me to close this issue? I realize it's not a feature that's enabled by default and I'm wondering whether we have sufficient documentation to indicate that it's an option. If not then I'm wondering what else we should put in place so that it's clear.

If you think we're good - feel free to just close this issue 😄

**Comment by aaronsteers** [comment]
I didn't see #751 prior to now, but for my part at least, I think that's perfect for my use cases. 👍

Thanks very much!

**Comment by barrywhart** [comment]
Seems good to me! Closing.
